### DISCUSSION TITLE
How can i create custom activity with dynamic outcome like fork activity for example ?

### DISCUSSION THREAD

**Discussion by hassanelsherbiny** [elsa-workflows/elsa-core#2580] [Q&A] [answered]
Im trying to create custom activity where i can control outcomes of it in designer just like fork activity.
i tried to use "IsDesignerCritical = true" attribute but it didn't work and i my activity only has Done Outcome.
also i have tried to copy the same code of fork activity from elsa code base but on designer it has only done outcome so what am i missing?
 
 this is my activity "Fork2"
![image](https://user-images.githubusercontent.com/20015837/145691883-b449f6fc-9fc5-4399-b0f6-bd25f12664fd.png)

![image](https://user-images.githubusercontent.com/20015837/145691871-c26acbc2-3bf9-4a13-8554-14c4203a2df3.png)

**Comment by sfmskywalker** [comment, marked as answer]
For this to work, you also need to implement a designer plugin that can dynamically add outcomes to the activity being rendered.

The documentation has the perfect example for your scenario: https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-designer-plugins#dynamic-outcomes

**Reply by hassanelsherbiny** [reply]
this is great and it worked with me but
how can i access "EventTypes" enum list to be able to register events and what is event type for activity property editor appear?
